src/actions/AccountReferralActions.js

```diff
@@ -199,8 +200,20 @@ async function validatePromoCards(account: EdgeAccount, cards: MessageTweak[]):
   let hasLinkedBankBool
 
   for (const card of cards) {
-    // Validate Country
+    // Validate app buildnum
+    const buildNum = getBuildNumber()
```

Comment:
In my stale promo banner PR I had a function to test version rather than build number which would be easier to use. Also, there's also the chance that build numbers may not be the same between iOS and android where version would be more consistent. Here's the function from the other PR:

export const testVersion = (deviceVersion: string, minVersion: string, maxVersion: string): boolean => {
  const [device, min, max] = [deviceVersion, minVersion, maxVersion].map(semver =>
    Number(
      semver
        .split('.')
        .map(id => id.padStart(2, '0'))
        .join()
        .replace(/,/g, '')
    )
  )

  return min <= device && device <= max
}

Reply:
Build number is more precise. The main purpose is so that we can do testing of a promo card and not hit customers with it, this let's us target an internal build number. also, if we intend to target customers, build numbers are always numerically increasing so it's easy to specify a range. semantic versions, while usable, aren't real numbers so ranges are awkward. I'm happy to also add a buildVersion for flexibility but I didn't think it was needed.